From 1c33633ef18274bf384c74c492195519be83c050 Mon Sep 17 00:00:00 2001
From: Julien Nabet <serval2412@yahoo.fr>
Date: Wed, 19 Feb 2014 21:23:58 +0000
Subject: Resolves: fdo#69925 Wiki Publisher Extension is not working

Regression from http://cgit.freedesktop.org/libreoffice/core/commit/?id=b80d8c695e6b6ded231e67ed9a80bcd1f1250c03
"compareTo" has been replaced by "==", so we compared the equality ot the pointers on the objects instead of the content of the strings
To make it simple and since we compare String objects, let's simply use "equals" method

Change-Id: I65bc1a87751297411fdbe722d3f49bce5e291638
(cherry picked from commit 9e2d1d21f9a58e320444f6eaea2f006530bbd2cb)
Signed-off-by: Michael Stahl <mstahl@redhat.com>
---
diff --git a/swext/mediawiki/src/com/sun/star/wiki/WikiEditorImpl.java b/swext/mediawiki/src/com/sun/star/wiki/WikiEditorImpl.java
index ff9f85a..4cad2f6 100644
--- a/swext/mediawiki/src/com/sun/star/wiki/WikiEditorImpl.java
+++ b/swext/mediawiki/src/com/sun/star/wiki/WikiEditorImpl.java
@@ -134,7 +134,7 @@ public final class WikiEditorImpl extends WeakBase
     {
         final com.sun.star.util.URL myURL = aURL;
         //logger.log( Level.INFO, "received dispatch request for: "+aURL.Complete );
-        if ( aURL.Protocol == protocolName )
+        if ( aURL.Protocol.equals(protocolName) )
         {
             /*
             synchronized( this )
@@ -146,7 +146,7 @@ public final class WikiEditorImpl extends WeakBase
 
             try
             {
-                if ( myURL.Path == "send" )
+                if ( myURL.Path.equals("send") )
                 {
                     sendArticle();
                 }
--
cgit v0.9.0.2-2-gbebe
